src/parsers/installrdf.js

```diff
@@ -1,66 +1,77 @@
-import { ADDON_TYPE_MAP, INSTALL_RDF, RDF_DEFAULT_NAMESPACE } from 'const';
-import { TYPE_INVALID, TYPE_MISSING } from 'messages';
+import { ADDON_TYPE_MAP, RDF_DEFAULT_NAMESPACE } from 'const';
+import { RDF_TYPE_INVALID, RDF_NAME_MISSING,
+         RDF_TYPE_MISSING } from 'messages';
 import log from 'logger';
-import RDFScanner from 'scanners/rdf';
 
 
 export default class InstallRdfParser {
 
-  constructor(rdfString, collector, {namespace=RDF_DEFAULT_NAMESPACE}={}) {
-    this.rdfString = rdfString;
+  constructor(xmlDoc, collector, {namespace=RDF_DEFAULT_NAMESPACE}={}) {
+    this.xmlDoc = xmlDoc;
     // Provides ability to directly add messages to
     // the collector.
     this.collector = collector;
     this.namespace = namespace;
   }
 
-  parseDoc() {
-    var rdfScanner = new RDFScanner(this.rdfString, INSTALL_RDF);
-    return rdfScanner.getContents();
-  }
-
   getMetaData() {
-    return this.parseDoc()
-      .then((xmlDoc) => {
```

Comment:
I like this simplification, but it introduces a required order of operations in the code now. `parseDoc` should cache the document and return it if it's already parsed (for performance), but always requesting the document means we never have to worry about when we call code. We do it elsewhere and it's handy for testing and other things.